osu.Game/Overlays/Profile/Header/PreviousUsernamesContainer.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Users;
+using osuTK;
+using System;
+using System.Linq;
+
+namespace osu.Game.Overlays.Profile.Header
+{
+    public class PreviousUsernamesContainer : Container
+    {
+        private const int duration = 200;
+        private const int margin = 10;
+        private const int width = 350;
+        private const int move_offset = 20;
+
+        public readonly Bindable<User> User = new Bindable<User>();
+
+        private readonly ContentContainer contentContainer;
+
+        public PreviousUsernamesContainer()
+        {
+            HoverIconContainer hoverIcon;
+
+            AutoSizeAxes = Axes.Y;
+            Width = width;
+            Children = new Drawable[]
+            {
+                contentContainer = new ContentContainer(),
+                hoverIcon = new HoverIconContainer(),
+            };
+
+            hoverIcon.ActivateHover += () =>
+            {
+                contentContainer.Show();
+                this.MoveToY(-move_offset, duration, Easing.OutQuint);
+            };
+
+            User.BindValueChanged(onUserChanged);
+            Hide();
+        }
+
+        private void onUserChanged(ValueChangedEvent<User> user)
+        {
+            contentContainer.Clear();
+
+            var usernames = user.NewValue.PreviousUsernames;
+
+            if (usernames.Any())
+            {
+                var amount = usernames.Length;
+
+                for (int i = 0; i < amount; i++)
+                {
+                    string text = (i + 1 == amount) ? usernames[i] : $@"{usernames[i]},";
+
+                    contentContainer.Usernames.AddText(new SpriteText
+                    {
+                        Font = OsuFont.GetFont(weight: FontWeight.Bold, italics: true),
+                        Text = text,
+                    });
+                }
```

Comment:
Rather than doing this whole stuff, do something like `Usernames.Text = string.Join(", ", usernames)`, and add the appropriate constructor argument on the text flow (`s => s.Font = OsuFont.GetFont(...)`).